doc/tutorials/enable_s5.rst

```diff
@@ -31,135 +31,202 @@ The diagram below shows the overall architecture:
+         VM_TYPE=service_vm
+         VM_NAME=Service_VM
+         DEV_NAME=tty:/dev/ttyS8,/dev/ttyS9,/dev/ttyS10,/dev/ttyS11,/dev/ttyS12,/dev/ttyS13,/dev/ttyS14
+         ALLOW_TRIGGER_S5=/dev/ttySn
+
+      .. note:: The mapping between User VM ID and communication serail device name (/dev/ttySn)
+                  in the file: `/etc/serial.conf`. If ``/dev/ttySn`` is configured in the  ``ALLOW_TRIGGER_S5``,
+                  this means system shutdown is allowed to be triggered in the corresponding User VM.
 
-   #. Copy ``life_mngr`` and ``life_mngr.service`` into the User VM:
+   #. Edit options in ``/etc/life_mngr/life_mngr.conf`` in the User VM.
 
       .. code-block:: none
 
-         $ scp build/misc/services/life_mngr root@<test board address>:/usr/bin/life_mngr
-         $ scp build/misc/services/life_mngr.service root@<test board address>:/lib/systemd/system/life_mngr.service
+         VM_TYPE=user_vm
+         VM_NAME=<User VM name>
+         DEV_NAME=tty:/dev/ttyS1
+         #ALLOW_TRIGGER_S5=/dev/ttySn
 
-   #. Use the below commands to enable ``life_mngr.service`` and restart the User VM.
+      .. note:: The User VM name in this configuration file should be consistent with the VM name in the
+               launch script for the Post-launched User VM or the VM name which is specified in the hypervisor
+               scenario XML for the Pre-launched User VM.
+
+   #. Use the following commands to enable ``life_mngr.service`` and restart the Service VM and User VMs.
 
       .. code-block:: none
 
-         # chmod +x /usr/bin/life_mngr
-         # systemctl enable life_mngr.service
-         # reboot
+         sudo chmod +x /usr/bin/life_mngr
+         sudo systemctl enable life_mngr.service
+         sudo reboot
+
+      .. note:: For the Pre-launched User VM, need restart Lifecycle Manager service manually
+         after Lifecycle Manager in Service VM starts.
 
 #. For the WaaG VM, run the lifecycle manager daemon:
 
-   a) Build the ``life_mngr_win.exe`` application::
+   a) Build the ``life_mngr_win.exe`` application and ``s5_trigger_win.py``::
+
+        cd acrn-hypervisor
+        make life_mngr
 
-        $ cd acrn-hypervisor
-        $ make life_mngr
+      .. note:: If there is no ``x86_64-w64-mingw32-gcc`` compiler, you can run
+               ``sudo apt install gcc-mingw-w64-x86-64`` on Ubuntu to install it.
 
-      .. note:: If there is no ``x86_64-w64-mingw32-gcc`` compiler, you can run ``sudo apt install gcc-mingw-w64-x86-64``
-         on Ubuntu to install it.
+   #) Copy ``s5_trigger_win.py`` into the WaaG VM.
 
    #) Set up a Windows environment:
 
-      I) Download the :kbd:`Visual Studio 2019` tool from `<https://visualstudio.microsoft.com/downloads/>`_,
+      1) Download the Python3 from `<https://www.python.org/downloads/release/python-3810/>`_, install
+         "Python 3.8.10" in WaaG.
+
+      #) Download the Visual Studio 2019 tool from `<https://visualstudio.microsoft.com/downloads/>`_,
```

Comment:
What do we need the Visual Studio tool for? This seems like a serious constraint for people who want to deploy a WaaG VM and enable S5 in it.

Reply:
If Lifecycle Manager for WaaG will be built in Windows, need to install Visual Studio tool. If Lifecycle Manager for WaaG is built in Linux, this tool is not needed. BTW, I will update document about this.